### DISCUSSION TITLE
Lesson 29, 30 in foundry fundamentals

### DISCUSSION THREAD

**Discussion by accurec** [Cyfrin/foundry-full-course-cu#2963] [Q&A] [answered]
Hello! I am following the lesson and able to start the node in Docker - getting this in terminal:
```
In memory node started v0.1.0-alpha.34:
 - zkSync Node (L2):
  - Chain ID: 260
  - RPC URL: http://127.0.0.1:8011
  - Rich accounts: https://docs.zksync.io/build/test-and-debug/in-memory-node#pre-configured-rich-wallets
 - Note: every restart will necessitate a reset of MetaMask's cached account data
 ```

Also can see the container running an image in Docker UI.

However, when I try to deploy contract to the RPC I get the following error:
```
Error: error sending request for url (http://127.0.0.1:8011/)

Context:
- Error #0: error sending request for url (http://127.0.0.1:8011/)
- Error #1: client error (SendRequest)
- Error #2: connection closed before message completed
```

Has anyone encountered something like this?

**Comment by heheboii11** [comment]
Please run anvil in your terminal
Open a new terminal than the one you are using and run anvil in that terminal 
just enter the command ```anvil``` in the new terminal

**Reply by accurec** [reply]
Hello! Yes I am running Anvil in separate terminal.

**Reply by heheboii11** [reply]
Can you close your vs code and rerun anvil and try to deploy again ? If you still face the issue please post the screenshot the issue you are facing

**Reply by accurec** [reply]
Tried to restart - seeing same issue.

<img width="1051" alt="Screenshot 2024-12-02 at 9 59 30 PM" src="https://github.com/user-attachments/assets/d3bd8000-9e9b-401a-b49a-7f7ec15f144b">

**Reply by NikolayPIvanov** [reply]
When you write `docker ps` is the port 8011 exposed to be called from your machine? Can you run a `ping` command against the `http://127.0.0.1:8011` and see if it responds?

Lastly, try to call `http://localhost:8011`, yeah it might seem dumb, but in some situations it works due to the network configuration

**Reply by heheboii11** [reply]
Lets wait for someone  who can solve this issue, I do not have knowledge in zksync development. May be try @NikolayPIvanov suggestion it might work

**Reply by accurec** [reply]
Hello! Unable to ping either address. In `docker ps` seems like the port is exposed:

<img width="1051" alt="Screenshot 2024-12-02 at 10 06 01 PM" src="https://github.com/user-attachments/assets/9e2abce3-9bdc-4e71-aa30-8f4ddcbaf6f7">

**Reply by EngrPips** [reply]
Patrick explicitly said we should not sweat over setting up a local `zksync` chain and just continue with the course if stuff didn't work.

**Reply by heheboii11** [reply]
yes, feel the same do not go too hard on zkysync, just use anvil or ethereum sepolia.

**Reply by EngrPips** [reply]
Indeed.

**Reply by accurec** [reply]
Yeah I know it is optional, but wanted to figure this one out regardless.

**Reply by EngrPips** [reply]
I get the picture. Please update us on the solution so that it will serve as a reference to help anyone who runs into the same issue in the future.

**Comment by fbozhkov** [comment, marked as answer]
I have the same problem. I think it has something with docker port mapping/forwarding but I can't seem to solve it. In the end I just followed the installation without the zksync cli and it worked https://docs.zksync.io/build/test-and-debug/in-memory-node#install-and-set-up-era_test_node

**Reply by EngrPips** [reply]
Awesome! Thanks for the heads-up, Friend.
